diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/ClassLikeStructuresOrderRule.kt

```diff
@@ -150,6 +124,47 @@ class ClassLikeStructuresOrderRule(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
+    /**
+     * When [this] is a ClassBody node, returns nested classes grouped by whether they are used inside [this] class
+     */
+    private fun ASTNode.getUsedAndUnusedClasses() = getChildren(TokenSet.create(CLASS))
+        .partition { classNode ->
+            classNode.getIdentifierName()?.let { identifierNode ->
+                parents()
+                    .last()
+                    .findAllDescendantsWithSpecificType(REFERENCE_EXPRESSION)
+                    .any { ref ->
+                        ref.parent({ it == classNode }) == null && ref.text.contains(identifierNode.text)
+                    }
+            } ?: false
+        }
+        .let { it.first.toMutableList() to it.second.toMutableList() }
+
+    /**
+     * @param blocks list of class elements with leading whitespaces and comments
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    private fun ASTNode.checkAndReorderBlocks(blocks: List<List<ASTNode>>) {
```

Comment:
do we need an assert or require in this method? If it is too dangerous, then do not add

Reply:
What sort of assert do you mean?

Reply:
by ASTNode type

Reply:
Added a not in KDocs